player/command.c

```diff
@@ -2769,6 +2770,20 @@ static int mp_property_osd_ass(void *ctx, struct m_property *prop,
     return m_property_read_sub(props, action, arg);
 }
 
+static int mp_property_term_size(void *ctx, struct m_property *prop,
+                                  int action, void *arg)
+{
+    int w = 80, h = 24;
```

Comment:
I prefer to fallback to 0,0, so that the caller can make different decisions when the terminal is 80x24 vs when the size is unknown.

Do check what other possibly-unknown numeric values fallback to, and if elsewhere uses `-1` for "unknown but returned anyway", then please use -1,-1 here too.

Reply:
0 would signify an error, we generally return `M_PROPERTY_UNKNOWN` when some properties are unknown

Reply:
Afaik `M_PROPERTY_UNAVAILABLE` is for when the property is currently not available, while M_PROPERTY_UNKNOWN is for properties that don't exist.

I do agree that we should return an error instead of a dummy value.

Reply:
Technically we can't know whether `get)terminal_size` failed or not, and we have a precedence that `osd-dimentions` has 0 if unknown. So I prefer 0,0.

Reply:
We can set w and h to -1 and then read the dimensions. I'm pretty sure a successful call would never return that.